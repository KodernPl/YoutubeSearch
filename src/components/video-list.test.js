import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import VideoList from './video-list';

const onVideoSelect = jest.fn();

beforeEach(() => {
  onVideoSelect.mockReset();
});

const renderComponent = () => {
  return shallow((
    <VideoList
      videos={[
        { etag: '1' },
        { etag: '2' },
      ]}
      onVideoSelect={onVideoSelect}
    />
  ));
};

test('renders correctly', () => {
  const component = renderComponent();

  expect(toJson(component)).toMatchSnapshot();
});
