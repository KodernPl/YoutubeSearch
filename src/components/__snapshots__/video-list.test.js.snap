// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<video-list__VideoListlWrapper>
  <VideoListItem
    key="1"
    onVideoSelect={[MockFunction]}
    video={
      Object {
        "etag": "1",
      }
    }
  />
  <VideoListItem
    key="2"
    onVideoSelect={[MockFunction]}
    video={
      Object {
        "etag": "2",
      }
    }
  />
</video-list__VideoListlWrapper>
`;
